src/go/rpk/pkg/cli/topic/describe_storage.go

```diff
@@ -0,0 +1,258 @@
+						humanReadable(r.CloudStatus.CloudLogSize, human, "size"),
+						humanReadable(r.CloudStatus.LocalLogSize, human, "size"),
+					)
+				}
+			})
+
+			header("SYNC", sync, withSections, func() {
+				isRrr := report[0].CloudStatus.CloudStorageMode == "read_replica"
+				headers := []string{"PARTITION", "LAST-SEGMENT-UPLOAD", "LAST-MANIFEST-UPLOAD", "METADATA-UPDATE-PENDING"}
+				if isRrr {
+					headers = append(headers, "LAST-MANIFEST-SYNC")
+				}
+				tw := out.NewTable(headers...)
+				defer tw.Flush()
+				for _, r := range report {
+					row := []any{
+						r.Partition,
+						humanReadable(r.CloudStatus.LastSegmentUpload, human, "time"),
+						humanReadable(r.CloudStatus.LastManifestUpload, human, "time"),
+						r.CloudStatus.MetadataUpdatePending,
+					}
+					// If isRrr is true, LastManifestSync should be there, this
+					// nil check is possibly redundant, but we want to avoid a
+					// nil pointer dereference.
+					if isRrr && r.CloudStatus.LastManifestSync != nil {
+						row = append(row, humanReadable(*r.CloudStatus.LastManifestSync, human, "time"))
+					}
+					tw.Print(row...)
+				}
+			})
+
+			header("LOG-SEGMENT-COUNT", count, withSections, func() {
+				tw := out.NewTable("PARTITION", "CLOUD", "LOCAL")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(r.Partition, r.CloudStatus.CloudLogSegmentCount, r.CloudStatus.LocalLogSegmentCount)
+				}
+			})
+
+			header("LOG-OFFSET", offset, withSections, func() {
+				tw := out.NewTable("PARTITION", "CLOUD-START", "CLOUD-LAST", "LOCAL-START", "LOCAL-LAST")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(r.Partition, r.CloudStatus.CloudLogStartOffset, r.CloudStatus.CloudLogLastOffset, r.CloudStatus.LocalLogStartOffset, r.CloudStatus.LocalLogLastOffset)
+				}
+			})
+		},
+	}
+	p.InstallAdminFlags(cmd)
+
+	cmd.Flags().BoolVarP(&all, "print-all", "a", false, "Print all cloud storage status")
+	cmd.Flags().BoolVarP(&summary, "print-summary", "s", false, "Print the summary section")
+	cmd.Flags().BoolVar(&size, "print-size", false, "Print the log size section")
+	cmd.Flags().BoolVar(&sync, "print-sync", false, "Print the sync section")
+	cmd.Flags().BoolVar(&count, "print-count", false, "Print the log count section")
+	cmd.Flags().BoolVar(&offset, "print-offset", false, "Print the offset section")
+	cmd.Flags().BoolVarP(&human, "human-readable", "H", false, "Print the times and bytes in a human-readable form")
+
+	return cmd
+}
+
+func getLastUpload(st []Status) int {
+	recent := st[0].CloudStatus.LastManifestUpload
+	for _, s := range st {
+		if s.CloudStatus.LastManifestUpload < recent {
+			recent = s.CloudStatus.LastManifestUpload
+		}
+		if s.CloudStatus.LastSegmentUpload < recent {
+			recent = s.CloudStatus.LastSegmentUpload
+		}
+	}
+	return recent
+}
+
+func humanReadable(v int, h bool, dataType string) any {
+	if !h {
+		return v
+	}
+	switch dataType {
+	case "size":
```

Comment:
`"size"` and `"time"` are used often enough here that I think they should be extracted to file level consts,

```
const (
    humanSize = "size"
    humanTime = "time"
)
```
which also makes it a lot easier to see that this switch is comprehensive